tests/modules/layers/test_anyprecision_optimizer.py

```diff
@@ -0,0 +1,165 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+
+from copy import deepcopy
+
+import pytest
+import torch
+import torch.nn as nn
+import torch.optim as optim
+
+from tests.test_utils import assert_expected, set_rng_seed
+from torchmultimodal.modules.optimizers.anyprecision import AnyPrecisionAdamW
+
+
+@pytest.fixture(autouse=True)
+def random():
+    set_rng_seed(2020)
+
+
+class TestAnyPrecisionOptimizer:
+    def _test_adam_equivalence(self, model, model_clone):
+        # Test non-default options
+        betas = (0.8, 0.88)
+        weight_decay = 0.03
+
+        adam_opt = optim.AdamW(
+            model_clone.parameters(), betas=betas, weight_decay=weight_decay
+        )
+        anyprecision_adam = AnyPrecisionAdamW(
+            model.parameters(),
+            variance_dtype=torch.float32,
+            betas=betas,
+            weight_decay=weight_decay,
+        )
+
+        # Verify params are equal initially
+        model_orig_params = [p.clone() for p in model.parameters()]
+        for p1, p2 in zip(model_clone.parameters(), model_orig_params):
+            assert_expected(p1, p2)
+
+        for i in range(6):
+            adam_opt.zero_grad()
```

Comment:
nit: can we test both settings of `set_to_none`?

Reply:
this is a great catch - yes I have updated the test to toggle based on even/odd of the index to use set_to_none = True/False. 
All passing.